src/eth/TransactionObject.js

```diff
@@ -116,7 +118,9 @@ export default class TransactionObject extends TransactionLifeCycle {
       this.setMined();
     } catch (err) {
       this._errorMessage = err.message;
-      this.setError(err);
+      console.error(err);
+      this._nonceService.setCounts();
```

Comment:
calling an async method here without await?

Reply:
I did this because I was thinking a user didn't really need to wait here for it to complete, since it would most likely be done by the time another transaction could be sent. But now that you mention it, in the edge case where it isn't done in time, it could cause some confusing errors. I'll go ahead and await it.